### PR TITLE
[MUL-288] Settlements payout

### PR DESCRIPTION
**Pull request by DominikWolos**

### CODE REVIEW

File: src/Component/Core/Common/StateMachine/SettlementStateMachineTransition.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public function __construct(
+        private FactoryInterface $productDraftStateMachineFactory,
+        private EntityManagerInterface $entityManager,
+    ) {
+    }
+
+    public function applyIfCan(
+        SettlementInterface $settlement,
+        string $transition,
+        bool $flush = false
```
Comment (senghe) [inline src/Component/Core/Common/StateMachine/SettlementStateMachineTransition.php:30]: To be honest, I don't get why we need the $flush parameter

File: src/Component/Core/Common/StateMachine/SettlementCallbacks.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+use BitBag\OpenMarketplace\Component\Settlement\Entity\SettlementInterface;
+
+final class SettlementCallbacks implements SettlementCallbacksInterface
+{
+    public function __construct(
+        private SettlementStateMachineTransitionInterface $settlementStateMachineTransition,
+    ) {
+    }
+
+    /**
```
Comment (senghe) [inline src/Component/Core/Common/StateMachine/SettlementCallbacks.php:24]: I would also put a special file in `doc/` directory where we would describe all the case.

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/Component/Core/Common/StateMachine/SettlementStateMachineTransition.php (hunk 1 above)
Comment (DominikWolos) [reply src/Component/Core/Common/StateMachine/SettlementStateMachineTransition.php:30]: I had some idea here, but it probably is worth removing.

File: src/Component/Core/Common/StateMachine/SettlementCallbacks.php (hunk 2 above)
Comment (DominikWolos) [reply src/Component/Core/Common/StateMachine/SettlementCallbacks.php:24]: Will do

File: doc/settlement_state_machine.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+##  Settlements
+
+### State Machine
```
Comment (senghe) [inline doc/settlement_state_machine.md:3]: The file currently is so technical. I would put there some content as "In order to use a payment gateway you need to handle..." etc.

**Review by senghe — CHANGES_REQUESTED** [review]

**Review by senghe — APPROVED** [review]

**Pull request merged**